frontend/console/src/features/timeline/details/TimelineIngressDetails.tsx

```diff
@@ -0,0 +1,96 @@
+import type { Timestamp } from '@bufbuild/protobuf'
+import { useContext, useEffect, useState } from 'react'
+import { AttributeBadge } from '../../../components/AttributeBadge'
+import { CloseButton } from '../../../components/CloseButton'
+import { CodeBlock } from '../../../components/CodeBlock'
+import type { IngressEvent } from '../../../protos/xyz/block/ftl/v1/console/console_pb'
+import { SidePanelContext } from '../../../providers/side-panel-provider'
+import { formatDuration } from '../../../utils/date.utils'
+import { DeploymentCard } from '../../deployments/DeploymentCard'
+import { verbRefString } from '../../verbs/verb.utils'
+import { TimelineTimestamp } from './TimelineTimestamp'
+
+export const TimelineIngressDetails = ({ timestamp, ingress }: { timestamp?: Timestamp; ingress: IngressEvent }) => {
+  const { closePanel } = useContext(SidePanelContext)
+  const [selectedIngress, setSelectedIngress] = useState(ingress)
+
+  useEffect(() => {
+    setSelectedIngress(ingress)
+  }, [ingress])
+
+  return (
+    <div className='p-4'>
+      <div className='flex items-center justify-between'>
+        <div className='flex items-center space-x-2'>
+          <div className=''>
+            {ingress.verbRef && (
+              <div className={'inline-block rounded-md bg-indigo-200 dark:bg-indigo-700 px-2 py-1 mr-1 text-sm font-medium text-gray-700 dark:text-gray-100'}>
+                {`${ingress.method} ${ingress.path}`}
+              </div>
+            )}
+          </div>
+          <TimelineTimestamp timestamp={timestamp} />
+        </div>
+        <CloseButton onClick={closePanel} />
+      </div>
+
+      <div className='text-sm pt-2'>Request</div>
+      <CodeBlock code={JSON.stringify(JSON.parse(selectedIngress.request), null, 2)} language='json' />
+
+      {selectedIngress.response !== 'null' && (
+        <>
+          <div className='text-sm pt-2'>Response</div>
+          <CodeBlock code={JSON.stringify(JSON.parse(selectedIngress.response), null, 2)} language='json' />
+        </>
+      )}
+
+      {selectedIngress.requestHeader !== 'null' && (
+        <>
+          <div className='text-sm pt-2'>Request Header</div>
+          <CodeBlock code={JSON.stringify(JSON.parse(selectedIngress.requestHeader), null, 2)} language='json' />
+        </>
+      )}
+
+      {selectedIngress.responseHeader !== 'null' && (
+        <>
+          <div className='text-sm pt-2'>Response Header</div>
+          <CodeBlock code={JSON.stringify(JSON.parse(selectedIngress.responseHeader), null, 2)} language='json' />
+        </>
+      )}
```

Comment:
Since these are so similar, what do you think about adding a helper component that looks like:
```
const CodeBlockIfNotNull({ heading, maybeNullContent }) => {
  if (maybeNullContent === 'null') {
    return
  }
  return (
    <>
      <div className='text-sm pt-2'>{heading}</div>
      <CodeBlock code={JSON.stringify(JSON.parse(maybeNullContent), null, 2)} language='json' />
    </>
  )
}
```

Reply:
Same below for the attribute badges

Reply:
I made a helper for the code blocks that are just `strings` but we'll want to rethink all of this with some better design so I don't want to spend too much effort on it just yet :)

Reply:
sweet sgtm!